measure-builds/src/test/java/com/automattic/android/measure/BuildTimePluginConfigurationCacheTests.kt

```diff
@@ -0,0 +1,129 @@
+
+import com.automattic.android.measure.models.BuildData
+import com.automattic.android.measure.models.ExecutionData
+import kotlinx.serialization.json.Json
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import java.io.File
+
+/**
+ * Test suite focused on validating the behavior of the Build Time Plugin when the configuration cache is enabled.
+ */
+class BuildTimePluginConfigurationCacheTests {
+
+    @BeforeEach
+    fun clear() {
+        File("build/functionalTest").listFiles()?.forEach {
+            if (it.name != "measure-builds") {
+                it.deleteRecursively()
+            }
+        }
+        File(".gradle/configuration-cache").deleteRecursively()
+    }
+
+    @Test
+    fun `given a project utilizes configuration cache, when build finishes, then report 0 configuration cache duration`() {
+        // given
+        val runner = runner()
+
+        // when
+        runner("help", "--stacktrace").build()
+
+        // then
+        assertThat(executionData.configurationPhaseDuration).isGreaterThan(0)
+
+        // when
+        runner("help").build()
+
+        // then
+        assertThat(executionData.configurationPhaseDuration).isEqualTo(0)
+    }
+
+    @Test
+    fun `given a project utilizes configuration cache, when build finishes, then assert that build or execution data was not reused`() {
+        fun runTaskAndGetDetails(vararg arguments: String): Pair<List<String>, Long> {
+            runner(*arguments).build()
+            return buildData.tasks to executionData.buildFinishedTimestamp
+        }
+
+        val (tasksA, timestampA) = runTaskAndGetDetails("help")
+        val (tasksB, timestampB) = runTaskAndGetDetails("outgoingVariants")
+        val (tasksC, timestampC) = runTaskAndGetDetails("tasks")
+
+        assertThat(timestampA).isNotEqualTo(timestampB)
+        assertThat(timestampA).isNotEqualTo(timestampC)
+        assertThat(timestampB).isNotEqualTo(timestampC)
+
+        assertThat(tasksA).isNotEqualTo(tasksB)
+        assertThat(tasksA).isNotEqualTo(tasksC)
+        assertThat(tasksB).isNotEqualTo(tasksC)
+    }
+
+    @Test
+    fun `given a project utilizes configuration cache, when build finishes twice with the same task, then assert that execution data was not reused`() {
+        runner("help").build()
+        val timestampA = executionData.buildFinishedTimestamp
+        runner("help").build()
+        val timestampB = executionData.buildFinishedTimestamp
+
+        assertThat(timestampA).isNotEqualTo(timestampB)
+    }
+
+    @Test
+    @Disabled(
+        "This test is the reason, why we have to be extra careful with the configuration cache. Build data won't always be invalidated." +
+            "This means that we can have a situation where the build data is, reused what is not expected. "
+    )
```

Comment:
Putting `prepareBuildData` outside of `project.afterEvaluate` doesn't change anything: bef5a2193f2bb9822c99607e3bd6f27a4a06cbe4

Reply:
More context: https://discuss.gradle.org/t/is-there-a-way-to-get-up-to-date-gradle-start-parameters-with-configuration-cache/47806